Fix -Wimplicit-function-declaration error with gcc 14.

Error:

```
In file included from /usr/include/ntfs-3g/inode.h:35,
                 from /usr/include/ntfs-3g/volume.h:51,
                 from /usr/include/ntfs-3g/device.h:33,
                 from ntfsclone-ng.c:33:
/usr/include/ntfs-3g/ntfstime.h: In function 'ntfs_current_time':
/usr/include/ntfs-3g/ntfstime.h:125:22: error: implicit declaration of function
'time' [-Wimplicit-function-declaration]
  125 |         now.tv_sec = time((time_t*)NULL);
      |                      ^~~~
```

--- partclone-0.3.32-origin/src/ntfsclone-ng.c
+++ partclone-0.3.32/src/ntfsclone-ng.c
@@ -25,6 +25,7 @@
 #include <sys/types.h>
 #include <linux/types.h>
 #include <errno.h>
+#include <time.h>

 #define NTFS_DO_NOT_CHECK_ENDIANS
 #define NTFS_MAX_CLUSTER_SIZE   65536
